lib/app/item.ex

```diff
@@ -235,16 +235,36 @@ defmodule App.Item do
     %{name: nil, num_items: 3, num_timers: 8, person_id: 0}
     %{name: username, num_items: 1, num_timers: 3, person_id: 1}
   ]
+
+  iex> person_with_item_and_timer_count(:person_id, :desc)
+  [
+    %{name: username, num_items: 1, num_timers: 3, person_id: 1}
+    %{name: nil, num_items: 3, num_timers: 8, person_id: 0}
+  ]
   """
-  def person_with_item_and_timer_count() do
+  def person_with_item_and_timer_count(
+        sort_column \\ :person_id,
+        sort_order \\ :asc
+      ) do
+    sort_column = to_string(sort_column)
+    sort_order = to_string(sort_order)
+
+    sort_column =
+      if validate_sort_column(sort_column), do: sort_column, else: "person_id"
+
+    sort_order = if validate_order(sort_order), do: sort_order, else: "asc"
+
     sql = """
     SELECT i.person_id,
     COUNT(distinct i.id) AS "num_items",
-    COUNT(distinct t.id) AS "num_timers"
+    COUNT(distinct t.id) AS "num_timers",
+    MIN(i.inserted_at) AS "first_inserted_at",
+    MAX(i.inserted_at) AS "last_inserted_at",
+    SUM(EXTRACT(EPOCH FROM (t.stop - t.start))) AS "total_timers_in_seconds"
     FROM items i
     LEFT JOIN timers t ON t.item_id = i.id
     GROUP BY i.person_id
-    ORDER BY i.person_id
+    ORDER BY #{sort_column} #{sort_order}
```

Comment:
These params should be `$1` and `$2` to avoid `SQL injection`
Then the `sort_column` and `sort_order` variables are passed in to the `SQL.query!/3`

Reply:
Take a look at my last comment on the other line.

I'm trying to build the query using Ecto.

Please, let me know what I can do.

Reply:
`Ecto` is lame for these kinds of complex queries. 🙃 
But I've since analysed your `validate_sort_column/1` and `validate_order/1` functions better (by adding tests for them) and am confident they mitigate `SQL` injection. ✅